pkg/controller/pxc/controller.go

```diff
@@ -969,7 +970,12 @@ func (r *ReconcilePerconaXtraDBCluster) deleteStatefulSet(cr *api.PerconaXtraDBC
 		return nil
 	}
 
-	err = r.client.Delete(context.TODO(), sfs.StatefulSet())
+	if !metav1.IsControlledBy(&sfsWithOwner, cr){
+		return nil
+	}
+
+	uid := sfsWithOwner.UID
```

Comment:
why do we assign it to separate variable?
We can use `&sfsWithOwner.UID` directly

Reply:
Updated accordingly